src/main/kotlin/com/hypto/iam/server/authProviders/WorkOSAuthProvider.kt

```diff
@@ -2,30 +2,54 @@ package com.hypto.iam.server.authProviders
 
 import com.hypto.iam.server.ROOT_ORG
 import com.hypto.iam.server.configs.AppConfig
+import com.hypto.iam.server.exceptions.EntityNotFoundException
+import com.hypto.iam.server.exceptions.UnknownException
 import com.hypto.iam.server.security.AuthMetadata
+import com.hypto.iam.server.security.AuthenticationException
 import com.hypto.iam.server.security.OAuthUserPrincipal
 import com.hypto.iam.server.security.TokenCredential
 import com.workos.WorkOS
+import com.workos.common.exceptions.NotFoundException
+import com.workos.common.exceptions.UnauthorizedException
+import mu.KotlinLogging
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
+import java.util.UUID
+
+private val logger = KotlinLogging.logger {}
 
 object WorkOSAuthProvider : BaseAuthProvider("workos"), KoinComponent {
     private val appConfig: AppConfig by inject()
+    private val workos: WorkOS by inject()
 
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
     override fun getProfileDetails(tokenCredential: TokenCredential): OAuthUserPrincipal {
-        val workos = WorkOS(appConfig.workOS.secretKey)
-        require(!tokenCredential.value.isNullOrEmpty()) {
-            "Token credential value cannot be null or empty"
+        return try {
+            if (tokenCredential.value.isNullOrEmpty()) {
+                throw UnauthorizedException("Token credential value cannot be null or empty", UUID.randomUUID().toString())
+            }
+            val profileAndToken = workos.sso.getProfileAndToken(tokenCredential.value, appConfig.workOS.clientId)
+            OAuthUserPrincipal(
+                tokenCredential,
+                ROOT_ORG,
+                profileAndToken.profile.email,
+                profileAndToken.profile.firstName ?: profileAndToken.profile.lastName ?: "",
+                "",
+                this.providerName,
+                AuthMetadata(profileAndToken.profile.connectionType + "-" + profileAndToken.profile.connectionId),
```

Comment:
AuthMetadata today has only 1 field: `id` which is meant for storing the id of the user provided by the authProvider. In this case, we'll have to map Jumpcloud userId to WorkOS attribute and then populate the value of that attribute here if at all we need this.

Do we really need this?

Reply:
We dont really need this. Thought of adding, idP connection type and connection id in metadata for debugging purpose. Will add profile id from workOS instead